cmd/notary/tuf.go

```diff
@@ -755,7 +757,8 @@ type passwordStore struct {
 }
 
 func (ps passwordStore) Basic(u *url.URL) (string, string) {
-	if ps.anonymous {
+	// if it's not a terminal, don't wait on input
+	if ps.anonymous || !terminal.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
IRL: leave as is for now as we use `os.Stdin` elsewhere too. Issue open already for looking into this.